Adding proxy configuration in the updates documentation (#327)
xcp-ng/xcp-ng-org: docs/management/updates.md

Your dom0 system must either have access to the internet, or to a local mirror. In the second case, make sure to update the `baseurl` values in `/etc/yum.repos.d/xcp-ng.repo` to make them point at the local mirror, and keep the mirror regularly synced.

#### Proxy configuration

> If your hosts need a proxy to access the repositories, there are a few ways to set it up depending on what you need:

Reviewer: ```suggestion
If your hosts require a proxy to access repositories, you have several options depending on your needs:
```

Author: Done.

For all yum repositories:
- Set `proxy=<proxy_url>` in the `[main]` section of `/etc/yum.conf`

For specific repositories:

> - Set `proxy=<proxy_url>` in the repository section of your choice in files present in `/etc/yum.repos.d`

Author: Done.

Set a system-wide proxy that will be used for everything, not only yum:
- Export the `http_proxy` and `https_proxy` variables in `/etc/environment`:
```bash
export http_proxy=<proxy_url>
export https_proxy=<proxy_url>
```

:::warning

> These changes will be kept alongside standard updates, but will be lost in case of an upgrade to a newer major version.

Reviewer: ```suggestion
These changes will remain intact through standard updates, but they will be lost if you upgrade to a new major version.
```

Author: Done.

Your dom0 system must either have access to the internet, or to a local mirror. In the second case, make sure to update the `baseurl` values in `/etc/yum.repos.d/xcp-ng.repo` to make them point at the local mirror, and keep the mirror regularly synced.

#### Proxy configuration

> If your hosts need to use a proxy to be able to access the repositories, there are multiple ways to configure it depending on your use case:

Reviewer: ```suggestion
If your hosts need a proxy to access the repositories, there are a few ways to set it up depending on what you need:
```

For all yum repositories:
- Set `proxy=<proxy_url>` in `[main]` section of `/etc/yum.conf`

For specific repositories:
- Set `proxy=<proxy_url>` in the repository section of your choice in files present in `/etc/yum.repos.d`

Set a system-wide proxy that will be used for everything, not only yum:

> - Add you can export lines for `http_proxy` and `https_proxy` variables in `/etc/environment`

Reviewer: ```suggestion
- Export the `http_proxy` and `https_proxy` variables in `/etc/environment`:
```

or

```suggestion
Add these export lines for `http_proxy` and `https_proxy` variables in `/etc/environment`
```

```bash
export http_proxy=<proxy_url>
export https_proxy=<proxy_url>
```

:::warning

> There are caveats for each methods:

Reviewer: ```suggestion
Each method has its own quirks:
```
or

```suggestion
Each method has its caveat:
```

> There are caveats for each methods:

Reviewer 2: Other caveats: none of this sticks after an upgrade to a newer version, AFAIK.

Author: I added a line in this callout to mention it (see last fixup).

- Updating the `yum` package may overwrite `/etc/yum.conf`
- Updating the `xcp-ng-release` package may overwrite files in `/etc/yum.repos.d/`

> - Updating the `setup` package may overwrite `/etc/environment`

Reviewer 2: Have you checked the way they are package, so that we can say for sure whether they will be overwritten or not? The counterpart when they're not overwritten is you don't get the latest changes and have to look for .rpmnew files (whereas overwritten changes are saved to a .rpmsave file).

Author: I only checked which package provides them, so no I'm not sure, I'll try to find out, but not all files are always singlely listed in spec files, so I need to understand what's the default behavior.

Author: After checking the spec files, it seem there is a `%config(noreplace)` directive for all these, so should I just remove these 3 lines and keep the warning about upgrades I added for your following comment?

Reviewer 2: Yes:
* they won't get evolutions of the file if updated via regular updates (the new packaged files being written as .rpmnew files instead), so that maintenance updates don't overwrite their settings
* on the contrary, when XCP-ng is upgraded to a newer version, they'll have to re-do their changes

Author: Fixed and reworded the sentence I about upgrades to make the distinction between updates and upgrades clear. Rebased on master as well.

> - Environmentn variables won't be taken into account when updating through Xen Orchestra

Reviewer: ```suggestion
- Environment variables won't be used when updating through Xen Orchestra
```
or
```suggestion
- Environment variables won't be taken into account when updating through Xen Orchestra
```

> - Environmentn variables won't be taken into account when updating through Xen Orchestra

Reviewer: ```suggestion
- Environment variables won't be used when updating through Xen Orchestra
```
or
```suggestion
- Environment variables won't be taken into account when updating through Xen Orchestra
```
